firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java

```diff
@@ -340,5 +372,30 @@ public void updateToLatestRelease_whenCheckForUpdateFails_updateAppNotCalled() t
     assertEquals(FirebaseAppDistributionException.Status.NETWORK_FAILURE, e.getErrorCode());
   }
 
-  // TODO: add updatetoLatestRelease tests with persistence logic when implemented
+  @Test
+  public void updateToLatestRelease_whenSignedIn_doesNotCallSignInTester() {
+    firebaseAppDistribution.onActivityResumed(activity);
+    when(mockCheckForUpdateClient.checkForUpdate())
+        .thenReturn(Tasks.forResult(TEST_RELEASE_NEWER_AAB_INTERNAL));
+    Mockito.doReturn(true).when(signInStorage).getSignInStatus();
+    firebaseAppDistribution.updateToLatestRelease();
+    verify(mockTesterSignInClient, never()).signInTester(any());
+  }
+
+  @Test
+  public void isTesterSignedIn_afterSuccessfulSignIn_returnsTrue() {
```

Comment:
Are these the tests that are requiring the spy instead of mock? Another way to test this is to assert that the storage method setSignInStatus is called with a true (or false) boolean.